docs: update end-of-life messages

jelbourn opened the pull request:
This change updates the messaging about AngularJS end-of-life in several places:
* Docs homepage
* Docs banner
* Security and support pages
* GitHub README

cc @aaronfrost 

@gkalpak am I missing anything here? I also want to update the changelog, but assume that happens automatically as part of the release process?

Splaktar reviewed (commented):
The commit message needs to be updated to something like the following to pass the lint checks:
```
docs(*): update end-of-life messages
```
I'm just posting the feedback that I was able to get through so far. I got pulled away and will have to finish reviewing/providing feedback later.

gkalpak commented on README.md:
```diff
@@ -14,9 +14,8 @@ piece of cake. Best of all? It makes development fun!
-**On July 1, 2018 AngularJS entered a 3 year Long Term Support period:** [Find out more](https://docs.angularjs.org/misc/version-support-status)
-
-**Looking for the new Angular? Go here:** https://github.com/angular/angular
+**[Support for legacy AngularJS has fully ended.](https://goo.gle/angularjs-end-of-life)**
+**Visit https://github.com/angular/angular for the actively supported Angular framework.**
```
In addition to that, I think it would also be useful to include the date when EOL started.

mgechev replied:
>**Learn more about [finding a path forward](https://blog.angular.io/finding-a-path-forward-with-angularjs-7e186fdd4429) for your AngularJS app.**

Makes sense to me

jelbourn replied:
Updated the text to this everywhere:
```markdown
AngularJS support has officially ended as of January 2022.
[See what ending support means](https://docs.angularjs.org/misc/version-support-status)
and [read the end of life announcement](https://goo.gle/angularjs-end-of-life).

Visit [https://angular.io](https://angular.io) for the actively supported Angular.
```

gkalpak reviewed (approved):
These are all the places that need updating afaict.

Regarding the changelog, you can manually update it if you want to add an EOL notice. Do we plan on having another release (and if so, what would it include)?

aaronfrost commented on SECURITY.md:
```diff
@@ -2,19 +2,13 @@
-| 1.3.x-1.7.x | :x:                |                       |                                      |
-| 1.2.x       | :warning:          | Security patches only | Last version to provide IE 8 support |
-| <1.2.0      | :x:                |                       |                                      |
-
```
I think it would be good to leave this table here, and have a date next to the 1.8.x and 1.2.x branches, showing when they ended. Removing this table entirely could leave the person wondering about the exact facts. If you leave this table here with EOL dates, it really drives home the point that it was supported until `12/31/2021`. 

I would also add a piece above the table that says something like:

`[AngularJS is no longer supported](https://goo.gle/angularjs-end-of-life). The following summarizes the End-of-life dates.`

aaronfrost commented on SECURITY.md:
```diff
@@ -2,19 +2,13 @@
-[0]: https://docs.angularjs.org/misc/version-support-status#long-term-support
+| Version     | Supported          | Status                | Comments                             |
+| ----------- | ------------------ | --------------------- | ------------------------------------ |
+| 1.8.x       | :x:                | All support ended     |                                      |
```
Add dates in the comments so that people have a little bit more context.

jelbourn replied:
I've added "as of January 2022" in the paragraph text above

jelbourn committed:
docs(*): update end-of-life messages

* Docs homepage
* Docs banner
* Security and support pages
* GitHub README
* CHANGELOG

jelbourn commented on docs/config/templates/app/indexPage.template.html:
```diff
@@ -146,7 +146,8 @@ <h4 class="search-results-group-heading">{{ key }}</h4>
             <p class="site-notice">
-              AngularJS is now in Long Term Support (LTS) mode: <a href="https://docs.angularjs.org/misc/version-support-status">Find out more</a>. Explore End Of Life (EOL) options <a href="https://blog.angular.io/finding-a-path-forward-with-angularjs-7e186fdd4429">here</a>.
+              <a href="https://goo.gle/angularjs-end-of-life">Support for legacy AngularJS has fully ended.</a>
+              Visit <a href="https://angular.io">https://angular.io</a> for the actively supported Angular.
```
The reason I put this here is because there's some subset of people that this part of the page and jump to the incorrect conclusion that this is referring to Angular. I'd like to keep this message everywhere to prevent that.

jelbourn commented on docs/config/templates/app/indexPage.template.html:
```diff
@@ -146,7 +146,8 @@ <h4 class="search-results-group-heading">{{ key }}</h4>
-              AngularJS is now in Long Term Support (LTS) mode: <a href="https://docs.angularjs.org/misc/version-support-status">Find out more</a>. Explore End Of Life (EOL) options <a href="https://blog.angular.io/finding-a-path-forward-with-angularjs-7e186fdd4429">here</a>.
+              <a href="https://goo.gle/angularjs-end-of-life">Support for legacy AngularJS has fully ended.</a>
+              Visit <a href="https://angular.io">https://angular.io</a> for the actively supported Angular.
             </p>
```
Updated the text to this everywhere:
```markdown
AngularJS support has officially ended as of January 2022.
[See what ending support means](https://docs.angularjs.org/misc/version-support-status)
and [read the end of life announcement](https://goo.gle/angularjs-end-of-life).

Visit [https://angular.io](https://angular.io) for the actively supported Angular.
```

jelbourn commented on docs/content/api/index.ngdoc:
```diff
@@ -5,7 +5,8 @@
-**On July 1, 2018 AngularJS entered a 3 year Long Term Support period:** [Find out more](misc/version-support-status).
+[Support for legacy AngularJS has fully ended](https://goo.gle/angularjs-end-of-life).
+Visit [https://angular.io](https://angular.io) for the actively supported Angular.
 </div>
```
I'd err towards keeping it- doesn't hurt to repeat ourselves.

jelbourn commented on docs/content/misc/version-support-status.ngdoc:
```diff
@@ -4,55 +4,9 @@
-You can read more about these plans in our [blog post announcement](https://blog.angular.io/stable-angularjs-and-long-term-support-7e077635ee9c).
-
-### Extended Long Term Support
+[Visit https://angular.io for the actively supported Angular framework](https://angular.io).
```
Added a summary here, PTAL

jelbourn commented on docs/content/misc/version-support-status.ngdoc:
```diff
@@ -4,55 +4,9 @@
-
-You can read more about these plans in our [blog post announcement](https://blog.angular.io/stable-angularjs-and-long-term-support-7e077635ee9c).
-
-### Extended Long Term Support
```
Added back the header (didn't actually mean to remove it in the first place)

Splaktar reviewed (approved):
LGTM other than the top spacing and overlay conflict on mobile.

jelbourn commented:
@Splaktar I've changed the top banner message to be inside the same container as the version picker to avoid the overlapping and updated the various `top` styles to adjust, seems okay now on various viewport sizes.  I also made the container opaque because I wouldn't ignore the transparency 😑

I've also changed that "CDN" text to "Release archive".

Splaktar commented on docs/config/templates/app/indexPage.template.html:
```diff
@@ -142,17 +142,19 @@ <h4 class="search-results-group-heading">{{ key }}</h4>
+              <a href="https://docs.angularjs.org/misc/version-support-status">
+                See what ending support means
+              </a> and
+              <a href="https://goo.gle/angularjs-end-of-life">
+                read the end of life announcement
+              </a>.<br>
```
```suggestion
              <a href="https://goo.gle/angularjs-end-of-life">read the end of life announcement</a>.<br>
```
This fixes the space before the period:

![Screen Shot 2022-03-10 at 02 16 32](https://user-images.githubusercontent.com/3506071/157609169-ccfb0bd0-a753-4920-b521-1fcd0c0bdb2c.png)

After:
![Screen Shot 2022-03-10 at 02 20 02](https://user-images.githubusercontent.com/3506071/157609340-d3939f02-c653-4112-8a93-fa8f249fd5b3.png)

Splaktar commented on docs/config/templates/app/indexPage.template.html:
```diff
@@ -142,17 +142,19 @@ <h4 class="search-results-group-heading">{{ key }}</h4>
+              <a href="https://goo.gle/angularjs-end-of-life">
+                read the end of life announcement
+              </a>.<br>
+              Visit <a href="https://angular.io">https://angular.io</a> for the actively supported
```
```suggestion
              Visit <a href="https://angular.io">angular.io</a> for the actively supported
```
The `https://` may be unnecessary in the banner, where space is somewhat limited.

jelbourn replied:
Removed `https://` from the link label 👍

mgechev reviewed (approved):
With the proposed changes by other reviewers, LGTM!
